testing/upload_validation.ts

```diff
@@ -988,6 +990,18 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
             return;
           }
         }
+
+        const propertyValueRaw = schema[propertyKey];
+        if (propertyValueRaw) {
+          if (Array.isArray(propertyValueRaw)) {
+            propertyValueRaw.forEach(propertyIdentifier => {
+              validatePropertyIdentifier(propertyIdentifier);
```

Comment:
Sorry for another nit 😦 I think this is making the error paths for these incorrect, since the index in the array is not take into account. The old code correctly had `path: ['subtitleProperties', i],`.

`validateJsonAndAssertFails` gives you back the error objects so you can make assertions on the path and message.